frame/support/src/traits/dispatch.rs

```diff
@@ -57,7 +57,7 @@ pub trait EnsureOrigin<OuterOrigin> {
 	/// ** Should be used for benchmarking only!!! **
 	#[cfg(feature = "runtime-benchmarks")]
 	fn try_successful_origin() -> Result<OuterOrigin, ()> {
-		Ok(Self::successful_origin())
+		unimplemented!();
```

Comment:
Hm how about deleting the default impl?  
How many compile errors does that cause? Having a panicing default impl isnt that useful either.

Reply:
Agree, it makes me (and probably any reader/user) wonder as well, why the default impl just calls the unimplemented macro.
In this particular use case the default impl is very useful because the users do not need to the successful_origin's functions implemented for all EnsureOrigin impls, only for the ones that really used in the benchmarks. So its a better experience.

Actually we can have Root origin returned as a default implementation.
Even if a user needs something different from the Root, and left the default impl, the user will see the meaningful error, and will be able to fix it quick.
What do you think?

Reply:
Okay I see. Then having unimplemented is probably better than root, so that devs will immediately see where they need to fix it.

Reply:
I think the actual better solution would be to remove `successful_origin`. IMO it was a failure to add this, the try variant makes much more sense. When a user for example has set `EnsureNever`, the benchmark should not even try to execute anything. Currently people may use `successful_origin` in their benchmark and then it fails. Then they need to fix their benchmark etc or request somewhere else that it gets fixed, while it would have been better from the beginning to directly use `try_successful_origin`

Reply:
Agreed. But if you remove it now then you have to fix all benches :see_no_evil:

Reply:
I removed `successful_origin` and returning `Err` as default impl for `try_successful_origin`.
If it looks good to you @ggwpez @bkchr , I will proceed with improving the PR description for these two changes and will create the companions.

Reply:
Ty @muharem :)

Reply:
Looks good to me.